examples/pony/passthrough/passthrough.pony

```diff
@@ -0,0 +1,275 @@
+/*
+
+Copyright 2019 The Wallaroo Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing
+ permissions and limitations under the License.
+
+*/
+
+use "buffered"
+use "options"
+use "wallaroo"
+use "wallaroo/core/common"
+use "wallaroo_labs/logging"
+use "wallaroo_labs/mort"
+use "wallaroo/core/sink/connector_sink"
+use "wallaroo/core/sink/tcp_sink"
+use "wallaroo/core/source"
+use "wallaroo/core/source/connector_source"
+use "wallaroo/core/source/tcp_source"
+use "wallaroo/core/state"
+use "wallaroo_labs/time"
+use "wallaroo/core/topology"
+
+type InputBlob is Array[U8] val
+
+actor Main
+  new create(env: Env) =>
+    Log.set_defaults()
+
+    var par_factor: USize = 64
+    let help = "<Wallaroo args, see below> [--help] [--verbose] [--parallelism] [--source-decoder framed|unframed] --source tcp|connector [--key-by first-byte|random] [--stage asis|asis-state|noop]* [--sink-parallelism] --sink tcp|connector"
+    var verbose: Bool = false
+    var source_decoder: FramedSourceHandler[InputBlob] val =
+      recover val InputBlobDecoder end
+
+    try
+      let pipeline = recover val
+        var source = recover ref
+          var src: (Pipeline[InputBlob] | None) = None
+          for o in recover ref make_options(env.args) end do
+            match o
+            | ("help", let arg: None) =>
+              @printf[I32]("Usage: $0 %s\n".cstring(), help.cstring())
+              StartupHelp()
+              Fail()
+            | ("verbose", None) =>
+              verbose = true
+            | ("parallelism", let p: U64) =>
+              par_factor = p.usize()
+            | ("source-decoder", let sd: String) =>
+              match sd
+                | "framed" =>
+                  source_decoder = recover val InputBlobDecoder end
+                | "unframed" =>
+                  source_decoder = recover val UnframedGreedyDecoder end
+                else
+                  @printf[I32]("Unknown source decoder: %s\n".cstring(),
+                    sd.cstring())
+                  Fail()
+                end
+            | ("source", let st: String) =>
+              match st
+              | "tcp" =>
+                src = Wallaroo.source[InputBlob]("Input",
+                  TCPSourceConfig[InputBlob].from_options(source_decoder,
+                    TCPSourceConfigCLIParser("InputBlobs", env.args)?
+                    where parallelism' = par_factor))
```

Comment:
Is it guaranteed these command line parameters will be parsed in the correct order?  This depends on "parallelism" having already been successfully parsed.  I think it might be safer to do the actual source config creation later, and use this to set a variable we use to pick.

Reply:
The value of `--parallelism` is checked in each of the three arg parsing loops, so when there are multiple `--parallelism` flags, will avoid using the last flag's value when creating a sink or step.  Also, the `--verbose` flag will help confirm that the source/step/sink is using the value that you expect.

Reply:
Does that mean that if it happens to parse "source" before "parallelism", you have to check the verbose output to see if it happened?

Reply:
In that case, yes.  There are rough edges here, so use `--verbose` when in doubt.

Reply:
My question is why are we doing it this way, instead of delaying creating the SourceConfig until after we parse the args, setting a temporary variable to indicate what kind of source config we'll eventually build?